### ISSUE TITLE
Put SugarBean emails in task queue and aggregate multiple to the same user

### ISSUE THREAD

**Issue by pcfreak30** [salesagility/SuiteCRM#7706]
#### Issue

Assignment notifications should be queued in the task system, and not sent out instantly per item. Additionally, multiple notifications should be batched together as a single email.

#### Expected Behavior

* updating user assignments should be instant without waiting on email
* multiple emails for the same user should be merged into 1 email to save on sending unneeded emails

#### Actual Behavior

You may end up waiting for up to 1 minute to assign 100 records to a single user

#### Possible Fix

Defer all sugarbean emails to the task queue, then keep a static array, keyed by the bean and user, to send all queued emails in memory at the end with a template that allows for many URLs to be linked.

#### Context

Lowering productivity by being forced to wait on email generation. It is a medium level issue I think.

#### Your Environment

* SuiteCRM Version used: 7.11.7
* Browser name and version: Version 75.0.3770.142 (Official Build) (64-bit)
* Environment name and version: Ubuntu/Docker/MariaDB/PHP7
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 16.04
